tempesta_fw/http.c

```diff
@@ -2606,6 +2669,24 @@ tfw_http_resp_fwd(TfwHttpResp *resp)
 	}
 }
 
+void
+tfw_h2_resp_adjust_fwd(TfwHttpResp *resp)
+{
+	TfwHttpReq *req = resp->req;
+	unsigned int stream_id;
+
+	if (!(stream_id = tfw_h2_stream_id_close(req))) {
+		tfw_http_resp_pair_free(req);
+		TFW_INC_STAT_BH(serv.msgs_otherr);
+		return;
+	}
+	/*
+	 * TODO: adjust, transform response into HTTP/2 and send it.
+	 */
```

Comment:
> join the conversation.

Well, it's been a while since the question was asked, but still.
I don't know what to add here. A single internal representation simplifies the code a lot, but an adaptive approach reduces conversion overhead. It's like good is good, and bad is bad. Can't really argue with that.